bson/src/main/org/bson/json/StrictCharacterStreamJsonWriter.java

```diff
@@ -182,23 +181,25 @@ public void writeString(final String value) {
     @Override
     public void writeRaw(final String value) {
         notNull("value", value);
-        checkPreconditions(State.VALUE);
+        checkState(State.VALUE);
         preWriteValue();
         write(value);
         setNextState();
     }
 
     @Override
     public void writeNull() {
-        checkPreconditions(State.VALUE);
+        checkState(State.VALUE);
         preWriteValue();
         write("null");
         setNextState();
     }
 
     @Override
     public void writeStartObject() {
-        checkPreconditions(State.INITIAL, State.VALUE);
+        if (state != State.INITIAL && state != State.VALUE) {
```

Comment:
Get rid of varargs to check pre-conditions.  This is the only place where there was more than one valid state